doc/rst/developer/bestPractices/ContributorInfo.rst

```diff
@@ -9,184 +9,274 @@ The git repository for the project is hosted on Github at
+
+* Iterate with the reviewer until you're both satisfied.  One should generally
+  try to do whatever their reviewer asks.  Sometimes, a reviewer will ask for
+  something really hard.  Try to make sure they understand the magnitude of the
+  request, and try to discuss if it's really necessary to do before merging.  If
+  you can't come to agreement, one of you should bring other developers
+  (individually or via chapel-developers_) into the conversation to get a
+  broader opinion (one of the jobs of the reviewer is to serve as a proxy for
+  other developers, or to bring those developers into the conversation if they
+  feel unqualified to do so).
+
+
+.. _Test changes:
+
+Test changes
+~~~~~~~~~~~~
 
-  .. note::
+If you did not have the resources to perform at least a full correctness test
+run, this is the point at which a reviewer would do so for you.  Contributors
+are expected to have verified any new tests work before asking a reviewer to do
+this.
 
-    Ideally, someone should volunteer to review your pull request within a day
-    or two. If this doesn't happen, feel free to make some noise. Ideally the
-    review should take place within a few days, though timing may vary
-    depending on other deadlines.
+Details on how to run tests may be found at `Test System`_, and details on the
+appropriate amount of testing before merging the final PR may be found at
+`Testing guidance`_.
 
-* Iterate with the reviewer until you're both satisfied. If you can't come to
-  agreement, one of you should bring other developers (individually or via
-  chapel-developers_) into the conversation to get a broader opinion (one of
-  the jobs of the reviewer is to serve as a proxy for other developers, or to
-  bring those developers into the conversation if they feel unqualified to do
-  so).
+.. _Getting ready to merge:
+
+Getting ready to merge
+~~~~~~~~~~~~~~~~~~~~~~
+
+Before the change can be merged, ensure:
+- all design changes have been discussed
+- the PR has been reviewed
+- the contributor agreement has been signed
+- the history in the PR does not contain anything not to be in the repo
+- relevant configurations pass testing
 
 * Once the pull request is approved, it can be merged. This can be done by
   either the reviewer or developer (given sufficient permissions), as decided
-  between the two of them.
+  between the two of them.  See `How to merge a PR`_ for steps to perform this.
+
+  Note: It's not generally possible to completely remove a commit from git by
+  the time it makes it in to the master branch. So be very careful not to commit
+  anything that you might regret later (e.g., sensitive code, code owned by
+  anyone other than yourself). Ideally, the review will catch such issues, but
+  the ultimate responsibility is on the developer.
+
+* If you are reviewing code from an external contributor without push
+  privileges, please look through the `Reviewer responsibilities`_ once more
+  before merging the change.
```

Comment:
My hope/intent is that the checklist above covers this part and would be easier for one of us to actually look through. If you agree, maybe refer to the checklist above?